### DISCUSSION TITLE
Promotional Popup

### DISCUSSION THREAD

**Discussion by bnhr** [radix-ui/primitives#1163] [Help] [answered]
Hi everyone,
Does anyone have an example of a promotional popup that doesn't require a button or trigger using Dialog Primitive?
Thanks

**Comment by jjenzz** [comment, marked as answer]
You can remove the `Dialog.Trigger` part and control it via the `open` or `defaultOpen` props: https://codesandbox.io/s/cranky-feather-n4ixe4?file=/src/App.js

**Reply by bnhr** [reply]
Thanks

**Comment by benoitgrelard** [comment]
I feel the urge to point out that this is really a dark pattern in terms of UI/UX but I'm sure we're all on the same wavelength here… 😀

**Reply by bnhr** [reply]
Yep, feel the same but...
